tests/test_protocols/test_base.py

```diff
@@ -107,3 +111,32 @@ def test_protocols_config(self):
             config=ProtocolConfig(),
         )
         assert protocol.config is not None
+
+
+class TestProtocolFromDir:
+    """Test the 'Protocol.from_dir' method."""
+
+    @classmethod
+    def setup_class(cls):
```

Comment:
I think we can remove the creation of temporary directory. But it is OK for future extensions.

Reply:
that's why I added it ;)..not because we need it lol